operator/v1/types_ingress.go

```diff
@@ -282,6 +287,178 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the schema for securing an IngressController.
+type SecuritySpec struct {
+	// profile defines the schema for a security profile.
+	Profile SecurityProfileSpec `json:"profile"`
+}
+
+// SecurityProfileSpec defines the schema for a security profile.
+// +union
+type SecurityProfileSpec struct {
+	// type is one of type OldSecurityProfile, IntermediateSecurityProfile,
+	// ModernSecurityProfile or CustomSecurityProfile. OldSecurityProfile,
+	// IntermediateSecurityProfile and ModernSecurityProfile profiles map to
+	// security compatiibility configurations defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// If unset, the IntermediateSecurityProfile type is used.
+	//
+	// +unionDiscriminator
+	// +optional
+	Type SecurityProfileType
+	// old is the "old" backward compatibility configuration as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	//
+	// +optional
+	// +nullable
+	Old *OldSecurityProfile
+	// intermediate is the "Intermediate" compatibility configuration (default) as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	// +nullable
+	Intermediate *IntermediateSecurityProfile
+	// modern is the "Modern" compatiibility configuration as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile
+	// custom defines the scehma of a user-defined custom compatiibility configuration.
```

Comment:
Typos, also what about "custom is a user-defined security profile."

Reply:
ack